jaxopt/_src/isotonic.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Isotonic Regression.
+
+Original code from:
+Fast Differentiable Sorting and Ranking
+Mathieu Blondel, Olivier Teboul, Quentin Berthet, Josip Djolonga
+In proceedings of ICML 2020 arXiv:2002.08871
```

Comment:
move this to somewhere inside the actual functions (cite it as "see section Y in Blondel et al.")

Reply:
Done